harness/asyncHelpers.js

```diff
@@ -0,0 +1,122 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+description: |
+    A collection of assertion and wrapper functions for testing asynchronous built-ins.
+defines: [asyncTest]
+---*/
+
+function asyncTest(testFunc) {
+  if (!Object.hasOwn(globalThis, "$DONE")) {
+    throw new Test262Error("asyncTest called without async flag");
+  }
+  if (typeof testFunc !== "function") {
+    $DONE(new Test262Error("asyncTest called with non-function argument"));
+    return Promise.resolve();
+  }
+  try {
+    return testFunc().then(
+      function () {
+        $DONE();
+      },
+      function (error) {
+        $DONE(error);
+      }
+    );
+  } catch (syncError) {
+    $DONE(syncError);
+    return Promise.resolve();
+  }
+}
+
+assert.throwsAsync = function (
+  expectedErrorConstructor,
+  funcOrThenable,
+  message
+) {
+  var innerThenable;
+  if (message === undefined) {
+    message = "";
+  } else {
+    message += " ";
+  }
+  if (typeof funcOrThenable === "function") {
+    try {
+      innerThenable = funcOrThenable();
+      if (
+        innerThenable === null ||
+        typeof innerThenable !== "object" ||
+        typeof innerThenable.then !== "function"
+      ) {
+        message +=
+          "Expected to obtain an inner promise that would reject with a" +
+          expectedErrorConstructor.name +
+          " but result was not a thenable";
+        return Promise.reject(new Test262Error(message));
```

Comment:
same here; why not throw this inside an async function?

Reply:
You're right, this one could and should be inside an async function - I see no reason it couldn't, anyway. I'll refactor, thanks! I think my brain was stuck on the asyncTest logic and I forgot to re-evaluate here.